demo/lightning/checkpoint/simulated/multiprocessing_train.py

```diff
@@ -176,53 +238,66 @@ def time_checkpoint_operation(benchmark_strategy: BenchmarkStrategy, distributed
             benchmark_strategy.save_checkpoint(
                 distributed_state_dict, filepath=checkpoint_path)
         elif operation == 'load':
-            benchmark_strategy.load_checkpoint(checkpoint_path=checkpoint_path)
+            benchmark_strategy.load_checkpoint(
+                checkpoint_path=checkpoint_path, initial_state_dict=template_state_dict)
         end_time = time.time()
         times.append(end_time - start_time)
         dist.barrier()  # Synchronize processes
         print(f"Completed iteration {i} for {operation} ...")
     return times
 
 
-def run_benchmark(rank, world_size: int, layer_size: int, project: str, filepath: str, padding_size: int, sample_count: int):
+def run_benchmark(rank, world_size: int, layer_size: int, project: str, filepath: str, padding_size: int, sample_count: int, debug: bool) -> None:
     setup(rank, world_size)
 
     model = SimpleModel(layer_size, padding_size)
 
-    dummy_input = torch.randn(100, layer_size)
-    _ = model(dummy_input)
-
-    full_state_dict = model.state_dict()
-    for i, tensor in enumerate(model.dummy_tensors):
-        full_state_dict[f'dummy_tensor_{i}'] = tensor
-
+    if rank == 0 and debug:
+        print("Writing initial model structure and parameters to file...")
+        write_full_model(model, "initial_model_state.txt")
     benchmark_strategy = BenchmarkStrategy(
         project=project, path=filepath, model=model)
 
-    distributed_state_dict = {f"{key}_shard_{rank}": split_tensor(
-        tensor, world_size, rank) for key, tensor in full_state_dict.items()}
+    state_dict = model.state_dict()
+    for i, tensor in enumerate(model.dummy_tensors):
+        state_dict[f'dummy_tensor_{i}'] = tensor
+
+    if rank == 0 and debug:
+        print("Writing state dict before saving to file...")
+        write_state_dict(state_dict, "state_dict_before_save.txt")
+        print("Shapes before saving:", {
```

Comment:
Is this printout massive? Do we want this?

Reply:
Not really massive, since I am only printing shape of each tensor. 
But it will be decent sized chunk as well. 
I think it is good to keep in the code (this will be later moved to benchmarks). So it can be used by devs with small model to check for correctness.